src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs

```diff
@@ -0,0 +1,163 @@
+            var validationMessages = _validator.ValidateOrganisationDetails(model);
+            if (validationMessages.Any())
+            {
+                model.ValidationErrors = new List<string>();
+                model.ValidationErrors.AddRange(validationMessages);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                return View("~/Views/Roatp/AddOrganisationDetails.cshtml", model);
+            }
+
+            _sessionService.SetAddOrganisationDetails(model);
+
+            return View("~/Views/Roatp/AddOrganisationPreview.cshtml", model);
+        }
+
+        [Route("successfully-added")]
+        public async Task<IActionResult> CreateOrganisation(AddOrganisationViewModel model)
+        {
+            var request = CreateAddOrganisationRequestFromModel(model);
+
+            var success = await _apiClient.CreateOrganisation(request);
+
+            if (!success)
+            {
+                model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                model.ValidationErrors = new List<string>
+                {
+                    $"An error occurred when adding the organisation '{model.LegalName}'.<br/> Please try again later."
+                };
+                return View("~/Views/Roatp/AddOrganisationPreview.cshtml", model);
+            }
+
+            var bannerModel = new BannerViewModel { CreateOrganisationCompanyName = model.LegalName };
+
+            return View("~/Views/Roatp/Index.cshtml", bannerModel);
+        }
+
+        [Route("back")]
+        public async Task<IActionResult> Back(string action, Guid organisationId)
+        {
+            var model = _sessionService.GetAddOrganisationDetails(organisationId);
+
+            return RedirectToAction(action, model);
+        }
+
+        private CreateOrganisationRequest CreateAddOrganisationRequestFromModel(AddOrganisationViewModel model)
+        {
+            var request = new CreateOrganisationRequest
+            {
+                Username = HttpContext.User.OperatorName(),
+                Organisation = CreateOrganisationFromModel(model)
+            };
+            return request;
+        }
+
+        private Organisation CreateOrganisationFromModel(AddOrganisationViewModel model)
+        {
+            var organisation = new Organisation
+            {
+                Id = Guid.NewGuid(),
+                LegalName = model.LegalName,
+                TradingName = model.TradingName,
+                OrganisationData = new OrganisationData
+                {
+                    CharityNumber = model.CharityNumber,
+                    CompanyNumber = model.CompanyNumber,
+                    FinancialTrackRecord = true,
+                    NonLevyContract = false,
+                    ParentCompanyGuarantee = false
+                },
+                UKPRN = Convert.ToInt64(model.UKPRN),
```

Comment:
This looks like it needs a TryParse

Reply:
Validation will be covered off in another story, APR-233 - anything non numeric or too small/large will be rejected before we get to here